src/pybind/mgr/dashboard/controllers/docs.py

```diff
@@ -0,0 +1,298 @@
+        elif isinstance(def_value, int):
+            return "integer"
+        return "string"
+
+    @classmethod
+    def _format_val(cls, val):
+        if isinstance(val, bool):
+            return "true" if val else "false"
+        elif isinstance(val, int):
+            return val
+        elif val is None:
+            return "null"
+        return '"{}"'.format(val)
+
+    @classmethod
+    def _gen_body_param(cls, body_params):
+        required = [p['name'] for p in body_params if p['required']]
+        required = ", ".join(['"{}"'.format(p) for p in required])
+
+        props = []
+        for p in body_params:
+            param_str = '"{}": {{"type": "{}"'.format(p['name'],
+                                                      cls._gen_type(p))
+            if 'default' in p:
+                param_str += ', "default": {}'.format(
+                    cls._format_val(p['default']))
+            param_str += "}"
+            props.append(param_str)
+        props = ", ".join(props)
+        if not props:
+            return None
+
+        return """{{
+            "in": "body",
+            "name": "body",
+            description: "",
+            "required": true,
+            "schema": {{
+                "type": "object",
+                "required": [{}],
+                "properties": {{
+                    {}
+                }}
+            }}
+        }}""".format(required, props)
+
+    @classmethod
+    def _gen_responses_descriptions(cls, method):
+        resps = []
+        if method.lower() == 'get':
+            resps.append('"200": {"description": "OK"}')
+        if method.lower() == 'post':
+            resps.append('"201": {"description": "Resource created."}')
+        if method.lower() == 'put':
+            resps.append('"200": {"description": "Resource updated."}')
+        if method.lower() == 'delete':
+            resps.append('"204": {"description": "Resource deleted."}')
+        if method.lower() in ['post', 'put', 'delete']:
+            resps.append('"202": {"description": "Operation is still executing.'
+                         ' Please check the task queue."}')
+        return """
+            "responses": {{
+                {},
+                "400": {{
+                    "description": "Operation exception. Please check the \
+response body for details."
+                }},
+                "401": {{
+                    "description": "Unauthenticated access. Please login first."
+                }},
+                "403": {{
+                    "description": "Unauthorized access. Please check your \
+permissions."
+                }},
+                "500": {{
+                    "description": "Unexpected error. Please check the \
+response body for the stack trace."
+                }}
+            }}
+        """.format(", ".join(resps))
+
+    @classmethod
+    def _gen_param(cls, param, ptype):
+        param_str = '"name": "{}", "in": "{}", "type": "{}"' \
```

Comment:
Why not using a dict for processing which is later converted into JSON? Should make it more easy to read.